libass/ass_parse.c

```diff
@@ -73,12 +73,14 @@ static inline void push_arg(struct arg *args, int *nargs, char *start, char *end
  */
 static inline int mystrcmp(char **p, const char *sample)
 {
-    int len = strlen(sample);
-    if (strncmp(*p, sample, len) == 0) {
-        (*p) += len;
+    char *p2;
+    for (p2 = *p; *p2 == *sample; p2++, sample++)
+        ;
```

Comment:
Won't this read past the legal memory region, if `{\tag` is at the very end of an event and safe memory region, and sample is also `tag`  ?
This seems to be fine in `parse_tags` since iinm there will always be atleast an `}` in `p`, but never in `sample`. This does however not apply to `event_has_hard_overrides`.

Reply:
I don't think that matters, since both strings will always be null-terminated?

Reply:
> since both strings will always be null-terminated?

Yes, which means `*p2 == *sample` is true, so both get incremented and dereferenced and read past their legal range.
Eg if both strings are `{'p', '\0'}` as in the above sample file.

Reply:
Ah! Good catch, fixing.